client/src/extension.ts

```diff
@@ -194,26 +198,37 @@ export function activate(context: ExtensionContext) {
     customCommands.switchImplIntf(client);
   });
 
-  // This sets up a listener that, if we're in dead code analysis mode, triggers
-  // dead code analysis as the LS server reports that ReScript compilation has
-  // finished. This is needed because dead code analysis must wait until
-  // compilation has finished, and the most reliable source for that is the LS
-  // server, that already keeps track of when the compiler finishes in order to
-  // other provide fresh diagnostics.
-  client.onReady().then(() => {
-    context.subscriptions.push(
-      client.onNotification("rescript/compilationFinished", () => {
-        if (inDeadCodeAnalysisState.active === true) {
-          customCommands.deadCodeAnalysisWithReanalyze(
-            inDeadCodeAnalysisState.activatedFromDirectory,
-            diagnosticsCollection,
-            diagnosticsResultCodeActions
-          );
-        }
-      })
-    );
+  commands.registerCommand("rescript-vscode.restart_language_server", () => {
+    client.stop().then(() => {
+      client = createLanguageClient();
+      attachDeadCodeAnalysis(client);

```

Comment:
This is done both here and below -- how about moving inside `createLanguageClient` so there's no chance to forget it.

Reply:
Also unsure why this:
```ts
commands.registerCommand("rescript-vscode.open_compiled", () => {
    customCommands.openCompiled(client);
  });
```
does not need to be done again after a restart.

And if it does not, how do we know it will be always the case. E.g. why it's not going to require the same treatment as `attachDeadCodeAnalysis` in future if it changes, and one might overlook to make the change here.

In general, is it possible to put all the logic in one place, and execute exactly the same logic at the beginning and after a restart?

Reply:
the above command works 
because the handler uses `client` which will always be the latest value because of closure.

Reply:
@cristianoc @zth I have made the change to attach deadcodeanalysis in the createLanguageServer function.